paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -152,6 +161,420 @@ void MatmulKernel(const Context &dev_ctx,
+  memory::desc CreateMemDescriptor(
+      const std::vector<int64_t> &dims,
+      funcs::OneDNNMemoryFormat format,
+      memory::data_type type = phi::funcs::OneDNNGetDataType<T>()) {
+    return phi::funcs::OneDNNMemDesc(dims, type, format);
+  }
+
+  template <typename T>
+  memory CreateMemory(const memory::desc &desc, const DenseTensor *tensor) {
+    return memory(
+        desc, engine_, phi::funcs::to_void_cast<T>(tensor->data<T>()));
+  }
+
+  memory CreateDstMemory(
+      const inner_product_forward::primitive_desc &mul_prim_desc,
+      const OneDNNContext &dev_ctx,
+      DenseTensor *output) {
+    auto dst_desc = mul_prim_desc.dst_desc();
+    auto buffer_size = dst_desc.get_size();
+
+    OT *output_data = dev_ctx.template Alloc<OT>(output, buffer_size);
+    output->set_mem_desc(dst_desc);
+    return memory(dst_desc, engine_, phi::funcs::to_void_cast<OT>(output_data));
+  }
+
+  memory Reorder(const memory::desc &src_desc,
+                 const memory::desc &dst_desc,
+                 void *src_data,
+                 void *dst_data = NULL) {
+    auto src_mem = memory(src_desc, engine_, src_data);
+    auto dst_mem = dst_data ? memory(dst_desc, engine_, dst_data)
+                            : memory(dst_desc, engine_);
+
+    auto reorder = dnnl::reorder(src_mem, dst_mem);
+
+    auto &astream = OneDNNContext::tls().get_stream();
+    {
+      paddle::platform::RecordEvent record_reorder(
+          "int_reorder",
+          paddle::platform::TracerEventType::UserDefined,
+          2,
+          paddle::platform::EventRole::kUniqueOp);
+      reorder.execute(astream, src_mem, dst_mem);
+      astream.wait();
+    }
+
+    return dst_mem;
+  }
+
+  memory TransposeInputY(const DenseTensor *input_y) {
+    auto dims = phi::vectorize<int64_t>(input_y->dims());
+    std::swap(dims[0], dims[1]);  // Correct output dimensions
+    auto src_desc =
+        CreateMemDescriptor<YT>(dims, funcs::OneDNNMemoryFormat::io);
+    auto dst_desc =
+        CreateMemDescriptor<YT>(dims, funcs::OneDNNMemoryFormat::oi);
+    return Reorder(
+        src_desc, dst_desc, phi::funcs::to_void_cast<YT>(input_y->data<YT>()));
+  }
+
+  const engine &engine_;
+  paddle::optional<memory> x_input_;
+  paddle::optional<memory> y_input_;
+  paddle::optional<memory> output_;
+  paddle::optional<inner_product_forward> mul_;
+  static constexpr bool is_int8_ = funcs::is_int8<XT>();
+};
+
+/* OT: output data type */
+template <typename XT, typename YT, typename OT>
+std::shared_ptr<MulPrimitiveFactory<XT, YT, OT>> GetPrimitiveFactory(
+    const OneDNNContext &dev_ctx,
+    const DenseTensor *input_x,
+    const DenseTensor *input_y,
+    const engine &onednn_engine) {
+  std::string key =
+      funcs::CreateKey(dev_ctx,
+                       paddle::framework::TransToProtoVarType(input_x->dtype()),
+                       phi::vectorize(input_x->dims()),
+                       paddle::framework::TransToProtoVarType(input_y->dtype()),
```

Comment:
We don't recommend using paddle::framework::TransToProtoVarType, you can try phi::TransToProtoVarType that is in phi/core/utils/data_type.h